library/src/main/java/org/apache/apex/malhar/lib/state/managed/Bucket.java

```diff
@@ -389,7 +389,7 @@ public void put(Slice key, long timeBucket, Slice value)
         sizeInBytes.getAndAdd(key.length);
         sizeInBytes.getAndAdd(Long.SIZE);
       }
-      if (timeBucket > bucketedValue.getTimeBucket()) {
+      if (timeBucket >= bucketedValue.getTimeBucket()) {
```

Comment:
Sorry for the delay. I think we need to check the time instead of timeBucketId.
Consider the below case, 
ManagedTimeStateImpl mt = new ManagedTimeStateImpl();
bucketSpanTime = 0.5 secs
mt.put(1004, k1, v1);
mt.put(1002, k1, v2);
Note: Both tuples are belong to same bucket.

Now, according to the fix, mt.get(k1).equals(v2) will return true.
Is it expected behavior. Please correct it, if I am wrong.